### DISCUSSION TITLE
Is it possible to run the task before the test is run?

### DISCUSSION THREAD

**Discussion by yuki2006** [golang/vscode-go#2108] [Help]
Before running the test from here, etc.

![image](https://user-images.githubusercontent.com/1004620/158417802-1e5d61e3-6fa3-4056-a336-85e02fc5b7f6.png)

I want to execute a task like "preLaunchTask" in launch.json.

Is it possible?

![image](https://user-images.githubusercontent.com/1004620/158418374-4c3138d5-15af-45c3-af44-b092b7711c83.png)

**Comment by hyangah** [comment]
Currently customizing the launch configuration of tests that run through test explorer UI is not supported.

But, I don't know if providing a vscode specific solution is a good idea.

In typical Go projects, developers expect that running `go test` command just works. From any editor, a terminal, ...
So, often times, people add the prerequisite tasks (e.g. environment setup) from the test code. Either, `MainTest`, helper function called from the beginning of each `Test.*` function, etc.

Or, another idea is ... is there any extension that implements Test UI API based on tasks.json? So, if users specify `go test` from tasks.json, that appear in the Test UI too.
